code/espurna/garland.cpp

```diff
@@ -0,0 +1,172 @@
+/*
+
+GARLAND MODULE
+
+Copyright (C) 2020 by Dmitry Blinov <dblinov76 at gmail dot com>
+
+*/
+
+#include "garland.h"
+#if GARLAND_SUPPORT
+
+#include <Adafruit_NeoPixel.h>
+
+#include "garland/anims.h"
+#include "garland/color.h"
+#include "garland/palette.h"
+#include "garland/scene.h"
+#include "web.h"
+#include "ws.h"
+
+const char* NAME_GARLAND_ENABLED        = "garlandEnabled";
+const char* NAME_GARLAND_BRIGHTNESS     = "garlandBrightness";
+
+const char* NAME_GARLAND_SWITCH         = "garland_switch";
+const char* NAME_GARLAND_SET_BRIGHTNESS = "garland_set_brightness";
+
+#define ANIMS                             8 //number of animations
```

Comment:
There is always `sizeof(anims) / sizeof(std::uintptr_t)`, in case we are using non-c++ containers and just plain arrays